File: RunSeasons/RunSeason2/backend/436.work_shedule/work_shed.go
// Превышает лимит времени 47 тест
package main

import (
	"fmt"
	"sort"
)

type Task struct {
	deadline int
	stress   int
}

func minimizeStress(n int, tasks []Task) int {
	// Сортируем задачи по стрессу в порядке убывания
	sort.Slice(tasks, func(i, j int) bool {
		return tasks[i].stress > tasks[j].stress
	})

	days := make([]bool, 200001) // Массив для отслеживания занятых дней
	totalStress := 0

	for _, task := range tasks {
		deadline := task.deadline
		stress := task.stress

		// Ищем первый свободный день от deadline до 1
		for deadline > 0 && days[deadline] {
			deadline--
		}
		if deadline > 0 {
			days[deadline] = true // Занимаем этот день
		} else {
			totalStress += stress // Если нет доступного дня, добавляем стресс
		}
	}

	return totalStress
}

func main() {
	var n int
	fmt.Scan(&n)

	tasks := make([]Task, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&tasks[i].deadline, &tasks[i].stress)
	}

	result := minimizeStress(n, tasks)
	fmt.Println(result)
}

File: competitions/RunSeason2/backend/436.work_shedule/work_shed.cpp
// Неправильный ответ 46 тест
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Task {
    int deadline;
    int stress;
};

int minimizeStress(int n, vector<Task>& tasks) {
    // Сортируем задачи по стрессу в порядке убывания
    sort(tasks.begin(), tasks.end(), [](const Task& a, const Task& b) {
        return a.stress > b.stress;
    });

    vector<bool> days(200001, false); // Массив для отслеживания занятых дней
    int totalStress = 0;

    for (const auto& task : tasks) {
        int deadline = task.deadline;
        int stress = task.stress;

        // Ищем первый свободный день от deadline до 1
        while (deadline > 0 && days[deadline]) {
            deadline--;
        }
        if (deadline > 0) {
            days[deadline] = true; // Занимаем этот день
        } else {
            totalStress += stress; // Если нет доступного дня, добавляем стресс
        }
    }

    return totalStress;
}

int main() {
    int n;
    cin >> n;

    vector<Task> tasks(n);
    for (int i = 0; i < n; i++) {
        cin >> tasks[i].deadline >> tasks[i].stress;
    }

    int result = minimizeStress(n, tasks);
    cout << result << endl;

    return 0;
}
